Realm/ObjectStore/impl/apple/external_commit_helper.cpp

```diff
@@ -211,79 +130,57 @@ ExternalCommitHelper::ExternalCommitHelper(Realm* realm)
         throw std::system_error(errno, std::system_category());
     }
 
-    // Create the anonymous pipe
-    int pipeFd[2];
-    ret = pipe(pipeFd);
+#else // !TARGET_OS_TV
+
+    // tvOS does not support named pipes, so use an anonymous pipe instead
+    int notification_pipe[2];
+    int ret = pipe(notification_pipe);
     if (ret == -1) {
         throw std::system_error(errno, std::system_category());
     }
 
-    m_shutdown_read_fd = pipeFd[0];
-    m_shutdown_write_fd = pipeFd[1];
+    m_notify_fd = notification_pipe[0];
+    m_notify_fd_write = notification_pipe[1];
 
-    // Use the minimum allowed stack size, as we need very little in our listener
-    // https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/Multithreading/CreatingThreads/CreatingThreads.html#//apple_ref/doc/uid/10000057i-CH15-SW7
-    pthread_attr_t attr;
-    pthread_attr_init(&attr);
-    pthread_attr_setstacksize(&attr, 16 * 1024);
+#endif // TARGET_OS_TV
 
-    auto fn = [](void *self) -> void * {
-        static_cast<ExternalCommitHelper *>(self)->listen();
-        return nullptr;
-    };
-    ret = pthread_create(&m_thread, &attr, fn, this);
-    pthread_attr_destroy(&attr);
-    if (ret != 0) {
+    // Create the anonymous pipe for shutdown notifications
+    int shutdown_pipe[2];
+    ret = pipe(shutdown_pipe);
+    if (ret == -1) {
         throw std::system_error(errno, std::system_category());
     }
-}
-
-ExternalCommitHelper::~ExternalCommitHelper()
-{
-    REALM_ASSERT_DEBUG(m_realms.empty());
-    notify_fd(m_shutdown_write_fd);
-    pthread_join(m_thread, nullptr); // Wait for the thread to exit
-}
-
-void ExternalCommitHelper::add_realm(realm::Realm* realm)
-{
-    std::lock_guard<std::mutex> lock(m_realms_mutex);
-
-    // Create the runloop source
-    CFRunLoopSourceContext ctx{};
-    ctx.info = realm;
-    ctx.perform = [](void* info) {
-        static_cast<Realm*>(info)->notify();
-    };
 
-    CFRunLoopRef runloop = CFRunLoopGetCurrent();
-    CFRetain(runloop);
-    CFRunLoopSourceRef signal = CFRunLoopSourceCreate(kCFAllocatorDefault, 0, &ctx);
-    CFRunLoopAddSource(runloop, signal, kCFRunLoopDefaultMode);
+    m_shutdown_read_fd = shutdown_pipe[0];
+    m_shutdown_write_fd = shutdown_pipe[1];
 
-    m_realms.push_back({realm, runloop, signal});
+    m_thread = std::async(std::launch::async, [=] {
+        try {
+            listen();
+        }
+        catch (std::exception const& e) {
+            fprintf(stderr, "uncaught exception in notifier thread: %s: %s\n", typeid(e).name(), e.what());
+            asl_log(nullptr, nullptr, ASL_LEVEL_ERR, "uncaught exception in notifier thread: %s: %s", typeid(e).name(), e.what());
```

Comment:
can you help me understand why the `fprintf` is necessary given that we're logging the same thing via `asl_log`?